### PR TITLE
[Hotfix] removing the empty familyName filter that prevents events from showing up

### PR DESCRIPTION
**Pull request by jeffreyliu3230**
Resolves [#OSF-4466].
Add a constraint so that the familyName filter is added only if the length of familyName of a contributor > 0.

### CODE REVIEW

File: website/static/js/share/results.js [hunk 1]
```diff
@@ -151,8 +151,10 @@ var Contributor = {
             m('a', {
                 href: '#',
                 onclick: function() {
-                    utils.updateFilter(vm, 'match:contributors.familyName:' + contributor.familyName, true);
                     utils.updateFilter(vm, 'match:contributors.givenName:' + contributor.givenName, true);
```
Comment (fabianvf) [inline website/static/js/share/results.js]: You should add the same constraint for givenName, and if neither are provided, make a filter for just the unparsed name. Long term, we can make that unanalyzed, and require it to be an exact match.

File: website/static/js/share/results.js [hunk 2]
```diff
@@ -151,8 +151,18 @@ var Contributor = {
             m('a', {
                 href: '#',
                 onclick: function() {
-                    utils.updateFilter(vm, 'match:contributors.familyName:' + contributor.familyName, true);
-                    utils.updateFilter(vm, 'match:contributors.givenName:' + contributor.givenName, true);
+                    var glen = contributor.givenName ? contributor.givenName.length : 0;
```
Comment (HarryRybacki) [inline website/static/js/share/results.js]: I think the variables should have more descriptive names such as`givenNameLength` or even `givenLength`

**Comment by HarryRybacki** [comment]
@jeffreyliu3230 -- looks good -- I'd just like to see more explicit vars.

**Comment by HarryRybacki** [comment]
Looks good to me CR1 passed :+1:

File: website/static/js/share/results.js [hunk 3]
```diff
@@ -151,8 +151,18 @@ var Contributor = {
             m('a', {
                 href: '#',
                 onclick: function() {
-                    utils.updateFilter(vm, 'match:contributors.familyName:' + contributor.familyName, true);
-                    utils.updateFilter(vm, 'match:contributors.givenName:' + contributor.givenName, true);
+                    var givenNameLength = contributor.givenName ? contributor.givenName.length : 0;
+                    var familyNameLength = contributor.familyName ? contributor.familyName.length : 0;
+                    if (givenNameLength <= 0 && familyNameLength <= 0) {
```
Comment (abought) [inline website/static/js/share/results.js]: Hopefully we'll never see a length < 0; perhaps use `=== 0` instead. 

You could alternately shorten this to  `if ( !givenNameLength && !familyNameLength)...` ...Though I admit to finding the numeric comparison a bit more readable.

**Comment by abought** [comment]
CR2 complete. No changes requested; one style comment added, but that's just bikeshedding.

Confirmed it worked for me running locally (using the scrapi lwbin harvester as test case).
